connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -291,6 +286,13 @@ private void maybeThrowProducerSendException() {
         }
     }
 
+    private void updateCommittableOffsets() {
+        Map<Map<String, Object>, Map<String, Object>> newOffsets = submittedRecords.committableOffsets();
+        synchronized (this) {
+            offsets.putAll(newOffsets);
+        }
```

Comment:
Yep, can do. I initially held off because there's only two synchronized blocks in the entire class now and they should both take very little time, but after checking out the parent `WorkerTask` it looks like we're doing some calls to the status listener in synchronized blocks as well. Seems worth the extra few lines here to avoid unnecessary synchronization. 👍